docs/contributing.md

```diff
@@ -0,0 +1,114 @@
+
+### As a MetaMask contributor
+
+If you're a MetaMask contributor, you can create these preview versions via draft pull requests:
+
+1. Navigate to your settings within GitHub and [create a classic access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic). Make sure to give this token the `packages:read` scope.
+2. Switch to your project locally and add a `.npmrc` file with the following content, filling in the appropriate areas:
+   ```
+   @metamask:registry=https://npm.pkg.github.com
+   //npm.pkg.github.com/:_authToken=<your personal access token>
+   ```
+   Make sure not to commit this file.
+3. Go to GitHub and open up a pull request for this repository, then post a comment on the PR with the text `@metamask-bot preview-build`. (This triggers the `publish-preview` GitHub action.)
+4. After a few minutes, you will see a new comment indicating that all packages have been published with the format `<package name>-<commit id>`.
+5. Switch back to your project locally and update `package.json` by replacing the versions for the packages you've changed in your PR using the new version format (e.g. `1.2.3-e2df9b4` instead of `~1.2.3`), then run `yarn install`.
+6. Repeat steps 3-5 after pushing new changes to your PR to generate and use new preview versions.
+
+### As an independent contributor
+
+If you're a contributor and you've forked this repository, you can create preview versions for a branch via provided scripts:
+
+1. Navigate to your settings within GitHub and [create a **classic** access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic). Make sure to give this token the `packages:read` scope.
+2. Switch to your project locally and add a `.npmrc` file with the following content, filling in the appropriate areas:
+   ```
+   @<your GitHub username>:registry=https://npm.pkg.github.com
+   //npm.pkg.github.com/:_authToken=<your personal access token>
+   ```
+   Make sure not to commit this file.
+3. Open the `package.json` for each package that you want to publish and change the scope in the name from `@metamask` to `@<your GitHub username>`.
+4. Switch to your fork of this repository locally and run `yarn prepare-preview-builds $(git rev-parse --short HEAD) && yarn build && yarn publish-previews` to generate preview versions for all packages based on the current branch and publish them to GitHub Package Registry. Take note of the version that is published; it should look like `1.2.3-e2df9b4` instead of `1.2.3`.
```

Comment:
Nit: Bash lint issue, always double quote to prevent string splitting (not sure it's possible for this command to return a space, but, :shrug: )

```suggestion
4. Switch to your fork of this repository locally and run `yarn prepare-preview-builds "$(git rev-parse --short HEAD)" && yarn build && yarn publish-previews` to generate preview versions for all packages based on the current branch and publish them to GitHub Package Registry. Take note of the version that is published; it should look like `1.2.3-e2df9b4` instead of `1.2.3`.
```

Reply:
Yup, good point.